geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisKeyCommandsFunctionExecutor.java

```diff
@@ -136,7 +136,11 @@ public void restore(RedisKey key, long ttl, byte[] data, RestoreOptions options)
   @Override
   public boolean rename(RedisKey oldKey, RedisKey newKey) {
     List<RedisKey> orderedKeys = orderForLocking(oldKey, newKey);
-    return stripedExecute(orderedKeys.get(0), () -> rename0(orderedKeys.get(1), oldKey, newKey));
+    return stripedExecute(orderedKeys.get(0), () -> {
+      // Trigger MOVED if necessary
+      getRedisData(newKey);
```

Comment:
I'm not sure I understand why this call of getRedisData(newKey) has been added but here are some questions/comments about it. Calling getRedisData(newKey) seems like a race. My understanding is that you are calling it so that it can throw an exception saying the bucket moved. But couldn't move right after you finish calling it and before rename0 is called? And since the get can go to the primary or secondary but the put that rename0 does on newKey goes to the primary isn't possible for that put to fail anyway? Also at this point of time we don't know that we have locked newKey. So if getRedisData(newKey) is needed shouldn't it be done in rename0 after both newKey and oldKey have been locked?

Reply:
`stripedExecute` calls `PartitionedRegion.computeWithPrimaryLocked` with the lambda defined here so the bucket shouldn't be able to move as part of this execution sequence.

Reply:
Jens and I talked about this and it turns out we now require that the two keys passed to rename are on the same primary bucket. That is why having done the first stripedExecute will have locked the primary even though the first stripedExecute may have been done on oldKey not newKey. We agreed that it would be better to do a check to ensure that both keys are on the same primary  and then this extra getRedisData(newKey) call is not needed.